logging/logger_test.go

```diff
@@ -45,3 +50,24 @@ func TestAddAndRemoveWriter(t *testing.T) {
 	assert.Equal(t, len(logger.unstructuredColorWriters), 0)
 	assert.Equal(t, len(logger.structuredWriters), 0)
 }
+
+func TestDisabledColors(t *testing.T) {
+	// Create a base logger
+	logger := NewLogger(zerolog.InfoLevel)
+
+	// Add colorized logger
+	var buf bytes.Buffer
+	logger.AddWriter(&buf, UNSTRUCTURED, true)
+
+	// We should expect the underlying data structures are correctly updated
+	assert.Equal(t, len(logger.unstructuredColorWriters), 1)
+
+	// Disable colors and log msg
+	colors.DisableColor()
+	logger.Info("foo")
+
+	// Ensure that msg doesn't include colors afterwards
+	prefix := fmt.Sprintf("%s %s", colors.LEFT_ARROW, "foo")
+	_, ok := strings.CutPrefix(buf.String(), prefix)
```

Comment:
So i'm not exactly sure if this tests whether colors are being shown or not? Note that `colors.LEFT_ARROW`, although in the `colors` package is not a color. LEFT_ARROW is a Unicode character. Colors, for terminal output, are actually defined as ANSI escape codes. You can learn more here: https://www.lihaoyi.com/post/BuildyourownCommandLinewithANSIescapecodes.html

So, this test would naturally pass because `logger.Info(foo)` would result in `-> foo` which is equivalent to the prefix. I would recommend changing `logger.Info(foo)` to `logger.Info(colors.Green, "foo", colors.Reset)`. In this way, "foo" should be colored green but since colors are disabled, the prefix should correctly be found without any ANSI escape codes.

Note, that you can test the negative case as well where you re-enable colors and assert that the prefix is _not_ found (aka `ok` is false). This can all be done in the same unit test.

Reply:
Okay, agree on the both. Resolving this comment.